tests/unit/behaviors/TranslatablePageTest.php

```diff
@@ -64,4 +65,52 @@ public function testAlternateLocale()
         $title_fr = $page->title;
         $this->assertEquals('titre francais', $title_fr);
     }
+    
+    public function testThemeScanner()
+    {
+        $scanner = new MessageScanner();
+
+        $check_strings = [
+            // Should not match
+            ["hello", 0],
+            ["'hello'|_", 0],
+            ["{ 'hello'|_ }", 0],
+            ["{{ var|_ }}", 0],
+            ["{{ 'hello'|upper|_ }}", 0],
+
+            // Code is syntactically wrong
+            ["{{ 'hello\"|_ }}", 0],
+            ["{{ \"hello'|_ }}", 0],
+
+            // Should find 1 match
+            ["{{ 'hello'|_ }}}}", 1],
+            ["{{{{ 'hello'|_ }}", 1],
+            ["{{ 'hello'|_ }}", 1],
+            ["{{ \"hello\"|_ }}", 1],
+            ["{{ \"'hello\"|_ }}", 1],
+            ["{{ '\"hello'|_ }}", 1],
+            ["{{ 'hello'|__ }}", 1],
+            ["{{ 'hello'|transRaw }}", 1],
+            ["{{ 'hello'|transRawPlural }}", 1],
+            ["{{ 'hello'|localeUrl }}", 1],
+            ["{{ 'hello'|_() }}", 1],
+            ["{{ 'hello'|_(func()) }}", 1],
+            ["{{ 'hello'|_({var: val}) }}", 1],
+            ["{{ 'hello'|_({var: func(param)}) }}", 1],
+            ["{{ 'hello'|_({var: func(nestedFunc())}) }}", 1],
+            ["{{ 'hello'|_|filter }}", 1],
+            ["{{ 'hello'|_|filter|otherfilter }}", 1],
+            ["{{ 'Apostrophe\'s'|_ }}", 1],
+            ['{{ "String with \"Double quote\""|_ }}', 1],
+
+            // Should find 2 matches
+            ['{{ \'Apostrophe\\\'s\'|_ }}{{ "String with \"Double quote\""|_ }}', 2],
+            ["{{ 'hello'|_|filter|otherfilter }}{{ 'hello'|_|filter|otherfilter }}", 2],
+            ["{{ 'hello'|transRaw('nested translation'|_) }}", 2],
+        ];
+
+        foreach ($check_strings as $check) {
+            $this->assertEquals($scanner->countMatches($check[0]), $check[1], 'String that failed: ' . $check[0]);
+        }
```

Comment:
@multiwebinc could you adjust the message scanner test to ensure that the correct messages are identified from the input strings?

Reply:
@LukeTowers Good idea.